src/hooks/UseIsRouteEnabledOnChain/index.tsx

```diff
@@ -0,0 +1,26 @@
+import { ChainId } from 'types';
+
+import { RouteName, routes } from 'constants/routing';
+import { useAuth } from 'context/AuthContext';
+
+export const allRoutes = Object.keys(routes) as (keyof typeof routes)[];
+export const disabledEthRoutes: typeof allRoutes = [
+  'corePool',
+  'corePoolMarket',
+  'convertVrt',
+  'history',
+  'vai',
+];
+const ethRoutes = allRoutes.filter(r => !disabledEthRoutes.includes(r));
+
+const enabledRoutesPerChain: Record<ChainId, RouteName[]> = {
+  [ChainId.BSC_MAINNET]: allRoutes,
+  [ChainId.BSC_TESTNET]: allRoutes,
+  [ChainId.ETHEREUM]: ethRoutes,
+  [ChainId.SEPOLIA]: ethRoutes,
+};
+
+export const useIsRouteEnabledOnChain = () => {
```

Comment:
How about using the existing `useIsFeatureEnabled` hook instead?
We could add a feature flag for each route:

```tsx
const featureFlags = {
   ...
   corePoolRoute: [ChainId. BSC_MAINNET, ChainId.BSC_TESTNET],
   corePoolMarketRoute:  [ChainId. BSC_MAINNET, ChainId.BSC_TESTNET],
   convertVrtRoute: [ChainId. BSC_MAINNET, ChainId.BSC_TESTNET],
   historyRoute: [ChainId. BSC_MAINNET, ChainId.BSC_TESTNET],
   vaiRoute: [ChainId. BSC_MAINNET, ChainId.BSC_TESTNET],
};
```